### PR TITLE
feat: populate payment method details in payments response

### PR DESCRIPTION
**Pull request by kashif-m**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Described in #5652 

### Additional Changes

- [x] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Helps in identification of payment method details which were used for the payment. This is returned in payment retrieve and payment webhooks.

## How did you test it?
<details>
  <summary>1. Complete and retrieve a gift card payment</summary>

    curl --location 'http://localhost:8080/payments' \
    --header 'Content-Type: application/json' \
    --header 'Accept: application/json' \
    --header 'api-key: dev_JCI3dRg06RjMyOrBDO81euxfOykGXlA9zgW4Cf2iXGy2xwexsY6GOSH5sI41GugN' \
    --data-raw '{
        "amount": 1100,
        "currency": "EUR",
        "confirm": true,
        "capture_method": "automatic",
        "capture_on": "2022-09-10T10:11:12Z",
        "amount_to_capture": 1100,
        "customer_id": "StripeCustomer",
        "email": "guest@example.com",
        "name": "John Doe",
        "phone": "999999999",
        "phone_country_code": "+65",
        "description": "Its my first payment request",
        "authentication_type": "no_three_ds",
        "return_url": "https://duck.com",
        "payment_method": "gift_card",
        "payment_method_type": "givex",
        "payment_method_data": {
            "gift_card": {
                "givex": {
                    "number": "6364530000000000",
                    "cvc": "122222"
                }
            }
        },
        "billing": {
            "address": {
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "city": "San Fransico",
                "state": "California",
                "zip": "94122",
                "country": "US",
                "first_name": "PiX"
            }
        },
        "shipping": {
            "address": {
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "city": "San Fransico",
                "state": "California",
                "zip": "94122",
                "country": "US",
                "first_name": "PiX"
            }
        },
        "statement_descriptor_name": "joseph",
        "statement_descriptor_suffix": "JS",
        "metadata": {
            "udf1": "value1",
            "new_customer": "true",
            "login_date": "2019-09-10T10:11:12Z"
        }
    }'
</details>

<details>
    <summary>2. Retrieve and verify in webhooks response as well</summary>

    curl --location 'http://localhost:8080/payments/pay_oRcnlCyFw1WVMQkg4fZq?force_sync=true' \
        --header 'Accept: application/json' \
        --header 'api-key: dev_p9yMRqUZKavXldvbcW8w0GfzR1ynBE2ORUu4FPY3rU82G8OFBx2XOq6ty71UrBIN'

Response

        {
            "payment_id": "pay_oRcnlCyFw1WVMQkg4fZq",
            "merchant_id": "postman_merchant_GHAction_011d2d37-e13d-4b83-a896-816c4d097d18",
            "status": "succeeded",
            "amount": 1100,
            "net_amount": 1100,
            "amount_capturable": 0,
            "amount_received": 1100,
            "connector": "adyen",
            "client_secret": "pay_oRcnlCyFw1WVMQkg4fZq_secret_NEFqOOcfDZyWNd5BXExn",
            "created": "2024-08-26T06:08:24.376Z",
            "currency": "EUR",
            "customer_id": "StripeCustomer",
            "customer": {
                "id": "StripeCustomer",
                "name": "John Doe",
                "email": "guest@example.com",
                "phone": "999999999",
                "phone_country_code": "+65"
            },
            "description": "Its my first payment request",
            "refunds": null,
            "disputes": null,
            "mandate_id": null,
            "mandate_data": null,
            "setup_future_usage": null,
            "off_session": null,
            "capture_on": null,
            "capture_method": "automatic",
            "payment_method": "gift_card",
            "payment_method_data": {
                "gift_card": {
                    "givex": {
                        "last4": "0000"
                    }
                },
                "billing": null
            },
            "payment_token": null,
            "shipping": {
                "address": {
                    "city": "San Fransico",
                    "country": "US",
                    "line1": "1467",
                    "line2": "Harrison Street",
                    "line3": "Harrison Street",
                    "zip": "94122",
                    "state": "California",
                    "first_name": "PiX",
                    "last_name": null
                },
                "phone": null,
                "email": null
            },
            "billing": {
                "address": {
                    "city": "San Fransico",
                    "country": "US",
                    "line1": "1467",
                    "line2": "Harrison Street",
                    "line3": "Harrison Street",
                    "zip": "94122",
                    "state": "California",
                    "first_name": "PiX",
                    "last_name": null
                },
                "phone": null,
                "email": null
            },
            "order_details": null,
            "email": "guest@example.com",
            "name": "John Doe",
            "phone": "999999999",
            "return_url": "https://duck.com/",
            "authentication_type": "no_three_ds",
            "statement_descriptor_name": "joseph",
            "statement_descriptor_suffix": "JS",
            "next_action": null,
            "cancellation_reason": null,
            "error_code": null,
            "error_message": null,
            "unified_code": null,
            "unified_message": null,
            "payment_experience": null,
            "payment_method_type": "givex",
            "connector_label": null,
            "business_country": null,
            "business_label": "default",
            "business_sub_label": null,
            "allowed_payment_method_types": null,
            "ephemeral_key": null,
            "manual_retry_allowed": false,
            "connector_transaction_id": "F7S9XBDCFJK94875",
            "frm_message": null,
            "metadata": {
                "udf1": "value1",
                "login_date": "2019-09-10T10:11:12Z",
                "new_customer": "true"
            },
            "connector_metadata": null,
            "feature_metadata": null,
            "reference_id": "pay_oRcnlCyFw1WVMQkg4fZq_1",
            "payment_link": null,
            "profile_id": "pro_gsiLSYq37GwBEhz0ev26",
            "surcharge_details": null,
            "attempt_count": 1,
            "merchant_decision": null,
            "merchant_connector_id": "mca_xR9f8EBOdN45KgAkTCLx",
            "incremental_authorization_allowed": null,
            "authorization_count": null,
            "incremental_authorizations": null,
            "external_authentication_details": null,
            "external_3ds_authentication_attempted": false,
            "expires_on": "2024-08-26T06:23:24.376Z",
            "fingerprint": null,
            "browser_info": null,
            "payment_method_id": null,
            "payment_method_status": null,
            "updated": "2024-08-26T06:08:25.813Z",
            "charges": null,
            "frm_metadata": null,
            "merchant_order_reference_id": null
        }

image
image

</details>

<details>
    <summary>3. Bank Debit - ACH</summary>

    curl --location 'http://localhost:8080/payments' \
    --header 'Content-Type: application/json' \
    --header 'Accept: application/json' \
    --header 'api-key: dev_JCI3dRg06RjMyOrBDO81euxfOykGXlA9zgW4Cf2iXGy2xwexsY6GOSH5sI41GugN' \
    --data-raw '{
        "amount": 10000,
        "currency": "USD",
        "confirm": true,
        "capture_method": "automatic",
        "capture_on": "2022-09-10T10:11:12Z",
        "customer_id": "StripeCustomer",
        "email": "abcdef123@gmail.com",
        "name": "John Doe",
        "phone": "999999999",
        "phone_country_code": "+65",
        "description": "Its my first payment request",
        "authentication_type": "three_ds",
        "return_url": "https://duck.com",
        "setup_future_usage": "on_session",
        "customer_acceptance": {
            "acceptance_type": "offline",
            "accepted_at": "1963-05-03T04:07:52.723Z",
            "online": {
                "ip_address": "127.0.0.1",
                "user_agent": "amet irure esse"
            }
        },
        "billing": {
            "address": {
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "city": "San Fransico",
                "state": "California",
                "zip": "94122",
                "country": "US"
            }
        },
        "browser_info": {
            "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
            "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
            "language": "nl-NL",
            "color_depth": 24,
            "screen_height": 723,
            "screen_width": 1536,
            "time_zone": 0,
            "java_enabled": true,
            "java_script_enabled": true,
            "ip_address": "127.0.0.1"
        },
        "shipping": {
            "address": {
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "city": "San Fransico",
                "state": "California",
                "zip": "94122",
                "country": "US",
                "first_name": "John",
                "last_name": "Doe"
            }
        },
        "statement_descriptor_name": "joseph",
        "statement_descriptor_suffix": "JS",
        "metadata": {
            "udf1": "value1",
            "new_customer": "true",
            "login_date": "2019-09-10T10:11:12Z"
        },
        "payment_method": "bank_debit",
        "payment_method_type": "ach",
        "payment_method_data": {
            "bank_debit": {
                "ach_bank_debit": {
                    "account_number": "40308669",
                    "routing_number": "121000358",
                    "sort_code": "560036",
                    "shopper_email": "example@gmail.com",
                    "card_holder_name": "joseph Doe",
                    "bank_account_holder_name": "David Archer",
                    "billing_details": {
                        "houseNumberOrName": "50",
                        "street": "Test Street",
                        "city": "Amsterdam",
                        "stateOrProvince": "NY",
                        "postalCode": "12010",
                        "country": "US",
                        "name": "A. Klaassen",
                        "email": "example@gmail.com"
                    },
                    "reference": "daslvcgbaieh"
                }
            }
        }
    }'
</details>

<details>
    <summary>4. Retrieve txn details</summary>

    curl --location 'http://localhost:8080/payments/pay_yyhXMuPPvXTjFYbFM4sS?force_sync=true' \
        --header 'Accept: application/json' \
        --header 'api-key: dev_p9yMRqUZKavXldvbcW8w0GfzR1ynBE2ORUu4FPY3rU82G8OFBx2XOq6ty71UrBIN'

Response

    {
        "payment_id": "pay_yyhXMuPPvXTjFYbFM4sS",
        "merchant_id": "postman_merchant_GHAction_011d2d37-e13d-4b83-a896-816c4d097d18",
        "status": "succeeded",
        "amount": 10000,
        "net_amount": 10000,
        "amount_capturable": 0,
        "amount_received": 10000,
        "connector": "adyen",
        "client_secret": "pay_yyhXMuPPvXTjFYbFM4sS_secret_8dripBlpBRYSlLwBD3qi",
        "created": "2024-08-26T06:10:37.599Z",
        "currency": "USD",
        "customer_id": "StripeCustomer",
        "customer": {
            "id": "StripeCustomer",
            "name": "John Doe",
            "email": "abcdef123@gmail.com",
            "phone": "999999999",
            "phone_country_code": "+65"
        },
        "description": "Its my first payment request",
        "refunds": null,
        "disputes": null,
        "mandate_id": null,
        "mandate_data": null,
        "setup_future_usage": "on_session",
        "off_session": null,
        "capture_on": null,
        "capture_method": "automatic",
        "payment_method": "bank_debit",
        "payment_method_data": {
            "bank_debit": {
                "ach": {
                    "account_number": "****8669",
                    "routing_number": "121***358",
                    "bank_account_holder_name": null,
                    "bank_name": null,
                    "bank_type": null,
                    "bank_holder_type": null
                }
            },
            "billing": null
        },
        "payment_token": null,
        "shipping": {
            "address": {
                "city": "San Fransico",
                "country": "US",
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "zip": "94122",
                "state": "California",
                "first_name": "John",
                "last_name": "Doe"
            },
            "phone": null,
            "email": null
        },
        "billing": {
            "address": {
                "city": "San Fransico",
                "country": "US",
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "zip": "94122",
                "state": "California",
                "first_name": null,
                "last_name": null
            },
            "phone": null,
            "email": null
        },
        "order_details": null,
        "email": "abcdef123@gmail.com",
        "name": "John Doe",
        "phone": "999999999",
        "return_url": "https://duck.com/",
        "authentication_type": "three_ds",
        "statement_descriptor_name": "joseph",
        "statement_descriptor_suffix": "JS",
        "next_action": null,
        "cancellation_reason": null,
        "error_code": null,
        "error_message": null,
        "unified_code": null,
        "unified_message": null,
        "payment_experience": null,
        "payment_method_type": "ach",
        "connector_label": null,
        "business_country": null,
        "business_label": "default",
        "business_sub_label": null,
        "allowed_payment_method_types": null,
        "ephemeral_key": null,
        "manual_retry_allowed": false,
        "connector_transaction_id": "L53M262VC5XBZX65",
        "frm_message": null,
        "metadata": {
            "udf1": "value1",
            "login_date": "2019-09-10T10:11:12Z",
            "new_customer": "true"
        },
        "connector_metadata": null,
        "feature_metadata": null,
        "reference_id": "pay_yyhXMuPPvXTjFYbFM4sS_1",
        "payment_link": null,
        "profile_id": "pro_gsiLSYq37GwBEhz0ev26",
        "surcharge_details": null,
        "attempt_count": 1,
        "merchant_decision": null,
        "merchant_connector_id": "mca_xR9f8EBOdN45KgAkTCLx",
        "incremental_authorization_allowed": null,
        "authorization_count": null,
        "incremental_authorizations": null,
        "external_authentication_details": null,
        "external_3ds_authentication_attempted": false,
        "expires_on": "2024-08-26T06:25:37.599Z",
        "fingerprint": null,
        "browser_info": {
            "language": "nl-NL",
            "time_zone": 0,
            "ip_address": "127.0.0.1",
            "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
            "color_depth": 24,
            "java_enabled": true,
            "screen_width": 1536,
            "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
            "screen_height": 723,
            "java_script_enabled": true
        },
        "payment_method_id": "pm_iBk2oopRt92TdFdZdLCH",
        "payment_method_status": "active",
        "updated": "2024-08-26T06:10:40.328Z",
        "charges": null,
        "frm_metadata": null,
        "merchant_order_reference_id": null
    }
        
image
image
</details>

<details>
  <summary>5. Bank Debit - Bacs</summary>

    curl --location 'http://localhost:8080/payments' \
      --header 'Content-Type: application/json' \
      --header 'Accept: application/json' \
      --header 'api-key: dev_JCI3dRg06RjMyOrBDO81euxfOykGXlA9zgW4Cf2iXGy2xwexsY6GOSH5sI41GugN' \
      --data-raw '{
          "amount": 10000,
          "currency": "USD",
          "confirm": true,
          "capture_method": "automatic",
          "capture_on": "2022-09-10T10:11:12Z",
          "customer_id": "StripeCustomer",
          "email": "abcdef123@gmail.com",
          "name": "John Doe",
          "phone": "999999999",
          "phone_country_code": "+65",
          "description": "Its my first payment request",
          "authentication_type": "three_ds",
          "return_url": "https://duck.com",
          "setup_future_usage": "on_session",
          "customer_acceptance": {
              "acceptance_type": "offline",
              "accepted_at": "1963-05-03T04:07:52.723Z",
              "online": {
                  "ip_address": "127.0.0.1",
                  "user_agent": "amet irure esse"
              }
          },
          "billing": {
              "address": {
                  "line1": "1467",
                  "line2": "Harrison Street",
                  "line3": "Harrison Street",
                  "city": "San Fransico",
                  "state": "California",
                  "zip": "94122",
                  "country": "US"
              }
          },
          "browser_info": {
              "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
              "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
              "language": "nl-NL",
              "color_depth": 24,
              "screen_height": 723,
              "screen_width": 1536,
              "time_zone": 0,
              "java_enabled": true,
              "java_script_enabled": true,
              "ip_address": "127.0.0.1"
          },
          "shipping": {
              "address": {
                  "line1": "1467",
                  "line2": "Harrison Street",
                  "line3": "Harrison Street",
                  "city": "San Fransico",
                  "state": "California",
                  "zip": "94122",
                  "country": "US",
                  "first_name": "John",
                  "last_name": "Doe"
              }
          },
          "statement_descriptor_name": "joseph",
          "statement_descriptor_suffix": "JS",
          "metadata": {
              "udf1": "value1",
              "new_customer": "true",
              "login_date": "2019-09-10T10:11:12Z"
          },
          "payment_method": "bank_debit",
          "payment_method_type": "ach",
          "payment_method_data": {
              "bank_debit": {
                  "ach_bank_debit": {
                      "account_number": "40308669",
                      "routing_number": "121000358",
                      "sort_code": "560036",
                      "shopper_email": "example@gmail.com",
                      "card_holder_name": "joseph Doe",
                      "bank_account_holder_name": "David Archer",
                      "billing_details": {
                          "houseNumberOrName": "50",
                          "street": "Test Street",
                          "city": "Amsterdam",
                          "stateOrProvince": "NY",
                          "postalCode": "12010",
                          "country": "US",
                          "name": "A. Klaassen",
                          "email": "example@gmail.com"
                      },
                      "reference": "daslvcgbaieh"
                  }
              }
          }
      }'
</details>

<details>
  <summary>6. Retrieve txn details</summary>

    curl --location 'http://localhost:8080/payments/pay_alnXBCvp6f9jl6ZqWo1e?force_sync=true' \
        --header 'Accept: application/json' \
        --header 'api-key: dev_p9yMRqUZKavXldvbcW8w0GfzR1ynBE2ORUu4FPY3rU82G8OFBx2XOq6ty71UrBIN'

Response

    {
        "payment_id": "pay_alnXBCvp6f9jl6ZqWo1e",
        "merchant_id": "postman_merchant_GHAction_011d2d37-e13d-4b83-a896-816c4d097d18",
        "status": "processing",
        "amount": 100,
        "net_amount": 100,
        "amount_capturable": 0,
        "amount_received": null,
        "connector": "adyen",
        "client_secret": "pay_alnXBCvp6f9jl6ZqWo1e_secret_dwQNXM4uPcs7PNeyxPnN",
        "created": "2024-08-26T06:11:42.510Z",
        "currency": "GBP",
        "customer_id": "StripeCustomer",
        "customer": {
            "id": "StripeCustomer",
            "name": "John Doe",
            "email": "abcdef123@gmail.com",
            "phone": "999999999",
            "phone_country_code": "+65"
        },
        "description": "Its my first payment request",
        "refunds": null,
        "disputes": null,
        "mandate_id": null,
        "mandate_data": null,
        "setup_future_usage": "on_session",
        "off_session": null,
        "capture_on": null,
        "capture_method": "automatic",
        "payment_method": "bank_debit",
        "payment_method_data": {
            "bank_debit": {
                "bacs": {
                    "account_number": "****8669",
                    "sort_code": "56**36",
                    "bank_account_holder_name": null
                }
            },
            "billing": null
        },
        "payment_token": null,
        "shipping": {
            "address": {
                "city": "San Fransico",
                "country": "US",
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "zip": "94122",
                "state": "California",
                "first_name": "John",
                "last_name": "Doe"
            },
            "phone": null,
            "email": null
        },
        "billing": {
            "address": {
                "city": "San Fransico",
                "country": "GB",
                "line1": "1467",
                "line2": "Harrison Street",
                "line3": "Harrison Street",
                "zip": "94122",
                "state": "California",
                "first_name": null,
                "last_name": null
            },
            "phone": null,
            "email": null
        },
        "order_details": null,
        "email": "abcdef123@gmail.com",
        "name": "John Doe",
        "phone": "999999999",
        "return_url": "https://duck.com/",
        "authentication_type": "no_three_ds",
        "statement_descriptor_name": "joseph",
        "statement_descriptor_suffix": "JS",
        "next_action": null,
        "cancellation_reason": null,
        "error_code": null,
        "error_message": null,
        "unified_code": null,
        "unified_message": null,
        "payment_experience": null,
        "payment_method_type": "bacs",
        "connector_label": null,
        "business_country": null,
        "business_label": "default",
        "business_sub_label": null,
        "allowed_payment_method_types": null,
        "ephemeral_key": null,
        "manual_retry_allowed": false,
        "connector_transaction_id": "FWJ6KDDCFJK94875",
        "frm_message": null,
        "metadata": {
            "udf1": "value1",
            "login_date": "2019-09-10T10:11:12Z",
            "new_customer": "true"
        },
        "connector_metadata": null,
        "feature_metadata": null,
        "reference_id": "pay_alnXBCvp6f9jl6ZqWo1e_1",
        "payment_link": null,
        "profile_id": "pro_gsiLSYq37GwBEhz0ev26",
        "surcharge_details": null,
        "attempt_count": 1,
        "merchant_decision": null,
        "merchant_connector_id": "mca_xR9f8EBOdN45KgAkTCLx",
        "incremental_authorization_allowed": null,
        "authorization_count": null,
        "incremental_authorizations": null,
        "external_authentication_details": null,
        "external_3ds_authentication_attempted": false,
        "expires_on": "2024-08-26T06:26:42.510Z",
        "fingerprint": null,
        "browser_info": {
            "language": "nl-NL",
            "time_zone": 0,
            "ip_address": "127.0.0.1",
            "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
            "color_depth": 24,
            "java_enabled": true,
            "screen_width": 1536,
            "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
            "screen_height": 723,
            "java_script_enabled": true
        },
        "payment_method_id": "pm_FWrcMK5kATtjyw9i0hqP",
        "payment_method_status": "inactive",
        "updated": "2024-08-26T06:11:48.748Z",
        "charges": null,
        "frm_metadata": null,
        "merchant_order_reference_id": null
    }

image
</details>

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

**Comment by Sarthak1799** [comment]
Payment method data changes look fine. 
Do get a review from payment-core.

**Review by SanchithHegde — APPROVED** [review]

File: crates/api_models/src/payments/additional_info.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+use crate::enums as api_enums;
+
+#[derive(Eq, PartialEq, Clone, Debug, serde::Deserialize, serde::Serialize, ToSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum BankDebitAdditionalData {
+    Ach(AchBankDebitAdditionalData),
+    Bacs(BacsBankDebitAdditionalData),
+    Becs(BecsBankDebitAdditionalData),
+    Sepa(SepaBankDebitAdditionalData),
+}
```
Comment (jarnura) [inline crates/api_models/src/payments/additional_info.rs:17]:
```suggestion
pub enum BankDebitAdditionalData {
    Ach(Box<AchBankDebitAdditionalData>),
    Bacs(Box<BacsBankDebitAdditionalData>),
    Becs(Box<BecsBankDebitAdditionalData>),
    Sepa(Box<SepaBankDebitAdditionalData>),
}
```

With Boxing the enum variant the memory allocation will be efficient.

File: crates/api_models/src/payments/additional_info.rs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    pub bank_name: Option<common_enums::BankNames>,
+    #[serde(flatten)]
+    pub additional_details: Option<BankRedirectDetails>,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, serde::Deserialize, serde::Serialize, ToSchema)]
+pub enum BankRedirectDetails {
+    BancontactCard(BancontactBankRedirectAdditionalData),
+    Blik(BlikBankRedirectAdditionalData),
+    Giropay(GiropayBankRedirectAdditionalData),
```
Comment (jarnura) [inline crates/api_models/src/payments/additional_info.rs:101]:
```suggestion
    BancontactCard(Box<BancontactBankRedirectAdditionalData>),
    Blik(Box<BlikBankRedirectAdditionalData>),
    Giropay(Box<GiropayBankRedirectAdditionalData>),
```

**Review by jarnura — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Review by Sarthak1799 — APPROVED** [review]
Payment method data changes LGTM

**Pull request merged**